### PR TITLE
telemetry: allow disable 128bit trace id for zipkin provider

### PR DESCRIPTION
**Pull request by zirain**
xref: https://github.com/istio/istio/issues/43359

Allows users to be compatible with existing zipkin backend.

### CODE REVIEW

**Comment by Stono** [comment]
👏 thank you for doing this!

File: mesh/v1alpha1/config.proto [hunk 1]
```diff
@@ -684,6 +684,10 @@ message MeshConfig {
       // Optional. Controls the overall path length allowed in a reported span.
       // NOTE: currently only controls max length of the path tag.
       uint32 max_tag_length = 3;
+
+      // Optional. A 128 bit trace id will be used in Istio.
+      // If true, will result in a 64 bit trace id being used.
+      bool Disable128BitTraceId = 4;
```
[the remark this answers is not in the document]
Comment (zirain) [reply mesh/v1alpha1/config.proto:690]: good idea!

File: mesh/v1alpha1/config.proto [hunk 2]
```diff
@@ -684,6 +684,10 @@ message MeshConfig {
       // Optional. Controls the overall path length allowed in a reported span.
       // NOTE: currently only controls max length of the path tag.
       uint32 max_tag_length = 3;
+
+      // Optional. A 128 bit trace id will be used in Istio.
+      // If true, will result in a 64 bit trace id being used.
+      bool Enable64BitTraceId = 4;
```
Comment (howardjohn) [inline mesh/v1alpha1/config.proto:690]: nit: use underscore casing like the rest.

Do we expect other formats to be present in the future? an ENUM may be more extensible if so?

Comment (zirain) [reply mesh/v1alpha1/config.proto:690]: for what I can see no, also I encourage people move to OpenTelemetry Tracing if there's no backport issue.

Comment (howardjohn) [reply mesh/v1alpha1/config.proto]: ok then this is probably fine

**Comment by zirain** [comment]
cc @kyessenov @lei-tang, PTAL

**Review by lei-tang — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Pull request merged**